mask/eolearn/mask/snow_mask.py

```diff
@@ -0,0 +1,261 @@
+        :type dem_params: (float, float)
+        :param red_params: Tuple specifying parameters to process the B04 red band. The first parameter defines the
+            scaling factor for down-sampling the red band, the second parameter is the maximum value of the
+            down-sampled red band for a dark cloud pixel, the third parameter is the minimum value
+            to return a non-snow pixel to the cloud mask, the fourth is the minimum reflectance value to pass the 1st
+            snow test, and the fifth is the minimum reflectance value to pass the 2nd snow test. With reference to the
+            ATBD, the tuple is (r_f, r_d, r_b, r_1, r_2)
+        :type red_params: (float, float, float, float, float)
+        :param ndsi_params: Tuple specifying parameters for the NDSI. First parameter is the minimum value to pass the
+            1st snow test, the second parameter is the minimum value to pass the 2nd snow test, and the third parameter
+            is the minimum snow fraction in the image to activate the pass 2 snow test. With reference to the
+            ATBD, the tuple is (n_1, n_2, f_s)
+        :type ndsi_params: (float, float, float)
+        """
+        super().__init__(data_feature, band_indices, **kwargs)
+        self.dem_feature = next(self._parse_features(dem_feature)())
+        self.clm_feature = next(self._parse_features(cloud_mask_feature)())
+        self.dem_params = dem_params
+        self.red_params = red_params
+        self.ndsi_params = ndsi_params
+        self.b10_index = b10_index
+        self._validate_params()
+
+    def _validate_params(self):
+        """ Check length of parameters defining threshold values """
+        for params, n_params in [(self.dem_params, 2), (self.red_params, 5), (self.ndsi_params, 3)]:
+            if not isinstance(params, (tuple, list)) or len(params) != n_params:
+                raise ValueError('Incorrect format or number of parameters for {0}. '
+                                 'Has to be a tuple of length {1}'.format(params, n_params))
+
+    def _resample_red(self, input_array):
+        """ Method to resample the values of the red band
+
+        The input array is first down-scaled using bicubic interpolation and up-scaled back using nearest neighbour
+        interpolation
+
+        :param input_array: input values
+        :return: resampled values
+        """
+        size = np.array(input_array.shape[1:]) // self.red_params[0]
+        resampled = [np.array(Image.fromarray(frame).resize(size, Image.BICUBIC).resize(frame.shape, Image.NEAREST))
+                     for frame in input_array]
+        return np.array(resampled)
+
+    def _adjust_cloud_mask(self, bands, cloud_mask, dem, b10):
+        """ Adjust existing cloud mask using cirrus band if L1C data and resampled red band """
+        # add to the existing cloud mask pixels found thresholding down-sampled red band and cirrus band/DEM
```

Comment:
This comment could probably go in a docstring 1 line above.

Reply:
added to docstring